### PR TITLE
call 'visibilityRules' method

### PR DESCRIPTION
**Pull request by tudor-malene**
### Why this change is needed

Part 3 of the visibility config implementation.
After https://github.com/ten-protocol/go-ten/pull/2057 and https://github.com/ten-protocol/go-ten/pull/2056

### What changes were made as part of this PR

- define the solidity interface
- call the `visibilityRules` method
- implement the config in the OBSErc20

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: go/enclave/evm/evm_facade.go [hunk 1]
```diff
@@ -196,7 +199,7 @@ func executeTransaction(
 
 		// Create a new context to be used in the EVM environment
 		blockContext := gethcore.NewEVMBlockContext(header, bc, author)
-		vmenv := vm.NewEVM(blockContext, vm.TxContext{BlobHashes: tx.Tx.BlobHashes(), GasPrice: header.BaseFee}, statedb, config, cfg)
+		vmenv = vm.NewEVM(blockContext, vm.TxContext{BlobHashes: tx.Tx.BlobHashes(), GasPrice: header.BaseFee}, statedb, config, cfg)
```
Comment (BedrockSquirrel) [inline go/enclave/evm/evm_facade.go:202]: Not really sure why this `applyTx` nested function exists, makes this quite hard to follow. But seems like vmenv will always be created or we'd have returned with error before you use it outside the function so guess it works.

**Review by BedrockSquirrel — COMMENTED** [review]
Seems reasonable to me but I'm not super familiar with the solidity usage.

File: go/enclave/evm/evm_facade.go (hunk 1 above)
Comment (tudor-malene) [reply go/enclave/evm/evm_facade.go:202]: tbh, I don't remember either. I think @StefanIliev545 might have worked in this area as well.
At some point we were depending on geth internal functions, which eventually became public

**Review by BedrockSquirrel — APPROVED** [review]
LGTM
